R/create_eq.R

```diff
@@ -25,35 +26,40 @@ create_eq.default <- function(lhs, rhs, ital_vars, use_coefs, coef_digits,
     rhs$final_terms <- add_greek(rhs, rhs$final_terms, greek, intercept, raw_tex)
   }
 
-  # Add error row
+  # Add error row or not in lm
+  if (!use_coefs){
   error_row <- rhs[nrow(rhs) + 1,]
   error_row$term <- "error"
   error_row$final_terms <- "\\epsilon"
-  rhs <- rbind(rhs, error_row)
+  rhs <- rbind(rhs, error_row)}
 
   list(lhs = list(lhs), rhs = list(rhs$final_terms))
 }
 
 #' @export
 #' @noRd
 #' @inheritParams extract_eq
-create_eq.glm <- function(lhs, rhs, ital_vars, use_coefs, coef_digits,
-                              fix_signs, model, intercept, greek, raw_tex) {
+create_eq.glm <- function(model, lhs, rhs, ital_vars, use_coefs, coef_digits,
+                              fix_signs, intercept, greek, raw_tex) {
   rhs$final_terms <- create_term(rhs, ital_vars)
 
   if (use_coefs) {
     rhs$final_terms <- add_coefs(rhs, rhs$final_terms, coef_digits)
   } else {
     rhs$final_terms <- add_greek(rhs, rhs$final_terms, greek, intercept, raw_tex)
   }
+  if (!is.null(model$offset)){
+    rhs <- rbind(rhs, c(rep(NA, (dim(rhs)[2]-1)),
+                        add_tex_ital(utils::tail(names(attr(model$terms, "dataClasses")),1), ital_vars)))
```

Comment:
In the equatiomatic-package.R file, can you add `#' @importFrom utils tail`, then run `devtools::document`, and then delete the namespacing here?

Reply:
Fixed.